File: js/templates/lead-pipeline/src/sync.ts
import type { Lead } from "./dedupe.js"

/** Implement this against your DB/CRM (Postgres, HubSpot, Salesforce, …). */
export interface SyncAdapter {
  name: string
  sync(leads: Lead[]): Promise<number>
}

/** Default stub: writes to stdout. Replace with a real adapter. */
export const consoleSync: SyncAdapter = {
  name: "console",
  async sync(leads: Lead[]): Promise<number> {
    for (const lead of leads) console.log("[sync]", lead.email)
    return leads.length
  },
}

File: python/templates/lead-pipeline/app/sync.py
from typing import Any, Dict, List, Protocol


class SyncAdapter(Protocol):
    """Implement against your DB/CRM (Postgres, HubSpot, Salesforce, …)."""

    name: str

    def sync(self, leads: List[Dict[str, Any]]) -> int: ...


class ConsoleSync:
    """Default stub: writes to stdout. Replace with a real adapter."""

    name = "console"

    def sync(self, leads: List[Dict[str, Any]]) -> int:
        for lead in leads:
            print("[sync]", lead.get("email"))
        return len(leads)


console_sync = ConsoleSync()
